File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int main(int argc, char* argv[])
{
    if (argc == 2 && atoi(argv[1]) > 0)
    {
        int key = atoi(argv[1]);
        
        while (key > 26)
        {
            key -= 26;
        }
        
        printf("plaintext: ");
        string plainText = get_string();
        printf("ciphertext: ");
        for (int i = 0, n = strlen(plainText); i < n; i++)
        {
            if (isalpha(plainText[i]) && isupper(plainText[i]))
            {
                if (plainText[i]+key > 'Z')
                {
                    printf("%c",plainText[i]+key-26);
                }
                else
                {
                     printf("%c", plainText[i]+key);
                }
               
            }
            else if (isalpha(plainText[i]) && islower(plainText[i]))
            {
                if (plainText[i]+key > 'z')
                {
                    printf("%c",plainText[i]+key-26);
                }
                else
                {
                     printf("%c", plainText[i]+key);
                }
               
            }
            else
            {
                printf("%c", plainText[i]);
            }
        }
        printf("\n");
    }
    else 
    {
        printf("wrong entry");
        return 1;
    }
}

File: pset6/caesar.py
import cs50
import sys
import string

#call the main function that shift the text, with the user input and the shift
def textShift(userInput, key):
    #turn the word into a list
    output = list(userInput)
    
    #interate over it and get the location and tha character
    for i, char in enumerate(output):
        # if character is a letter and is capatilized
        if char.isalpha() and char.isupper():
            # use the existing list, at the existing position and shift the origial letter
            # with the shifte letter that wraps around 
            output[i] = chr((ord(char) - ord('A') + key) % 26 + ord('A'))
        # do the same thing for the lower case letter
        elif char.isalpha() and char.islower():
            output[i] = chr((ord(char) - ord('a') + key) % 26 + ord('a'))
        else:
            # and if it is no letter just add the symbol
            output[i] = char
    # join all the list items into a string and return it
    code = ''.join(output)
    return code
    
# check if the user entered a second statement after the .py
if sys.argv[1] == False:
    print("command line: python caesar.py int > 0")
    sys.exit(1)

# call the main program and return the changed word in the next line
print(textShift(input("plainText: "), int(sys.argv[1])))

